Automattic/jetpack
# Videopress Block: Fix video width and lack of alignment classes

Fixes https://github.com/Automattic/wp-calypso/issues/46889
Fixes #17925 

VideoPress block on WordPress.com doesn't get the alignment CSS classes applied to it on save. This leads to the video appearing to be different widths in editor vs frontend.

#### Changes proposed in this Pull Request:
* Update application of CSS class names it to pick up the alignment CSS classes.
* Fix block attributes to prevent block validation errors that were occurring in production
* Move setting of `classNames` attribute out of the `render()` function to avoid `UnsavedChangesWarning` error.
* Add deprecation for older blocks that were missing alignment classes or didn't get `classNames` saved due to bug.


#### Does this pull request change what data or activity we track or use?
No.

#### Unit test for attribute migrations
1. Run `yarn test-extensions`
2. Confirm `extensions/blocks/videopress/test/index.js` ran successfully.

#### Testing instructions - Local:

1. Using master branch, start up local Jetpack site with sufficient plan to upload videos to videopress
2. Add a video block to a post and upload a video
3. Set it's alignment to `wide` and add a custom CSS class name via sidebar > advanced controls
4. Save post and view on frontend. (Should be missing `alignwide` CSS class)
5. Reload post in the editor, the block's will be invalid due to current bugs.
5. Checkout this PR and rebuild Jetpack
6. With dev tools console open, reload the editor and confirm no errors and the block deprecation is successfully applied
7. Save the post and reload the frontend. 
8. Confirm the block displays with wide alignment

#### Testing instructions - Sandbox:

1. Apply this PR branch and test VideoPress block functions correctly locally.
2. On a sandboxed site with a sufficient plan, add a video to a post, set its alignment to "wide", then save.
3. Confirm on the frontend that the video's block does not contain the `alignwide` CSS class.
4. Apply this PR's associated diff ( D53903-code ) or use a8c-wp-env and sync Jetpack to your sandbox.
5. Reload the editor for your sandboxed site and confirm the block is still valid.
6. Change the block's alignment to "full" and save post.
7. Switch to the frontend, reload and confirm the video block has the `alignfull` CSS class.

#### Screenshots

| Wide Align - Before | Wide Align - After |
|------------------------|-----------------------|
| <img width="943" alt="Screen Shot 2020-12-08 at 6 32 41 pm" src="https://user-images.githubusercontent.com/60436221/101462874-7408c780-3988-11eb-9a72-ba30be7e151d.png"> | <img width="944" alt="Screen Shot 2020-12-08 at 6 33 42 pm" src="https://user-images.githubusercontent.com/60436221/101462900-79661200-3988-11eb-8699-c6c5c8920668.png"> |

#### Proposed changelog entry for your changes:
* Fixes missing alignment classes on VideoPress block.
* Fixes VideoPress block validation errors
